extensions/interactions/ItemSelectionInput/ItemSelectionInput.js

```diff
@@ -27,30 +27,56 @@ oppia.directive('oppiaInteractiveItemSelectionInput', [
       scope: {},
       templateUrl: 'interaction/ItemSelectionInput',
       controller: ['$scope', '$attrs', function($scope, $attrs) {
-        $scope.items = oppiaHtmlEscaper.escapedJsonToObj($attrs.itemsWithValue);
-        var allowableSelections = $attrs.selectionCountWithValue;
+        $scope.choices = oppiaHtmlEscaper.escapedJsonToObj($attrs.choicesWithValue);
+        var maxAllowableSelectionCount = $attrs.maxAllowableSelectionCountWithValue;
+        var minAllowableSelectionCount = $attrs.minAllowableSelectionCountWithValue;
         $scope.studentSelections = {};
 
-        for (var i = 0; i < $scope.items.length; i++) {
-          $scope.studentSelections[$scope.items[i]] = false;
+        if (maxAllowableSelectionCount == 1) {
+          $scope.isCheckbox = false;
+        } else {
+          $scope.isCheckbox = true;
+        }
+
+
+        for (var i = 0; i < $scope.choices.length; i++) {
+          $scope.studentSelections[$scope.choices[i]] = false;
         }
 
+        $scope.answers = [];
+        $scope.maxed = false; // Indicates if the count is over max.
+        $scope.min = false;  // Indicates if count is under min.
+
+        $scope.checkCount = function(selection) {
+          index = $scope.answers.indexOf(selection);
+          if (index !== -1) {
+            $scope.answers.splice(index, 1);
+          } else if (!$scope.maxed) {
+            $scope.answers.push(selection);
+          }
+          var size = $scope.answers.length;
+          if (size >= maxAllowableSelectionCount) {
+            $scope.maxed = true;
+          } else {
+            $scope.maxed = false;
```

Comment:
Instead of setting these here, you could change $scope.maxed and $scope.min to be functions. E.g.

    $scope.notEnoughSelections = function() {
      // use JavaScript's reduce() to count the number of keys whose corresponding value is true, then
      //   return true if this is less than minAllowableSelectionCount
      // e.g., see http://stackoverflow.com/a/15748853
    };